tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,463 @@
+message means that `~nus` failed to connect to `~zod`'s `%shoe` session.
+
+### `%shoe`'s code
+
+```hoon
+::  shoe: example usage of /lib/shoe
+::
+::    the app supports one command: "demo".
+::    running this command renders some text on all sole clients.
+::
+/+  shoe, verb, dbug, default-agent
+```
+
+`/+` is the Ford rune which imports libraries from the `/lib` directory into
+the subject.
+ * `shoe` is the `shoe` library.
+ * `verb` is a library used to print what a Gall agent is doing.
+ * `dbug` is a library of debugging tools. Why do we need this?
+ * `default-agent` contains a Gall agent core with minimal implementations of
+   required Gall arms.
+
+```hoon
+|%
++$  state-0  [%0 ~]
++$  command  ~
+::
++$  card  card:shoe
+--
+```
+The types used by the app.
+
+`$state-0` stores the state of the app, which is null as there is no state to
+keep track of. It is good practice (required?) to have a type for state anyways
+in case the app is made stateful at a later time, and indeed we will do this
+when we modify this app later in the tutorial.
+
+`$command` is typically a set of tagged union types that are possible
+commands that can be entered by the user. Since this app only supports one
+command, it is unnecessary for it to have any associated data, thus the command
+is represented by `~`.
+
+In a non-trivial context, a `$command` is given by `[%name
+data]`, where `%name` is the identifier for the type of command and `data` is
+a type or list of types that contain data needed to execute the command. See
+`app/chat-cli.hoon` for examples of commands, such as `[%say letter:store]` and
+`[%delete path]`.
+
+`$card` is either an ordinary Gall agent `card` or a `%shoe` `card`, which takes
+the form `[%shoe sole-ids=(list @ta) effect=shoe-effect]`. A `%shoe` `card` is
+sent to all `sole`s listed in `sole-ids`, instructing them to implement the
+action specified by `effect`. Here we can
+reference `card:shoe` because of `/+  shoe` at the beginning of the app.
+
+```hoon
+=|  state-0
+=*  state  -
+::
+```
+Pin the bunt value of `state-0` to the head of the subject, then give it the
+macro `state`. The `-` here is a lark expression referring to the head of the
+subject. This allows us to use `state` to refer to the state elsewhere in the
+code no matter what version we're using.
+
+```hoon
+%+  verb  |
+%-  agent:dbug
+^-  agent:gall
+%-  (agent:shoe command)
+^-  (shoe:shoe command)
+```
+This sequence of commands forges a `agent:gall` core that
+possess may arms of the sort defined in `verb,` `agent:dbug`, and `agent:shoe`.
+This sequence of commands tells the compiler that the following core is a Gall
+agent core that may possess arms of the sort defined in `verb`, `agent:dbug`,
+and `agent:shoe`. These cores tell the compiler what sort of form these arms
+must have, and ensure that the arms that are not standard Gall arms are put into
+the context so that the resulting core nests within `agent:gall`.
```

Comment:
There's duplicate text here, and this section is hard to read regardless.

What's happening here is just a sequence of function calls, where we call the "agent wrapper" functions (for lack of a better term) for all those libraries we imported.

The explanation below is much clearer.

Reply:
https://github.com/urbit/docs/pull/912/commits/cae94e6de188b62721bb40412c37bf52b0aca6fc#diff-6d8f36f0b87617684df85577b6a7144dL298-R295

Reply:
Hmmm that didn't seem to link directly to where I wanted it to go...

Reply:
Single lines seem to work though. Apologies for my lack of GH-fu

https://github.com/urbit/docs/pull/912/commits/cae94e6de188b62721bb40412c37bf52b0aca6fc#diff-6d8f36f0b87617684df85577b6a7144dL298